src/Aspire.Dashboard/Model/ResourceGraph/ResourceGraphMapper.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Xml.Linq;
+using Aspire.Dashboard.Otlp.Model;
+using Aspire.Dashboard.Resources;
+using Microsoft.Extensions.Localization;
+using Microsoft.FluentUI.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components.Extensions;
+using Icons = Microsoft.FluentUI.AspNetCore.Components.Icons;
+
+namespace Aspire.Dashboard.Model.ResourceGraph;
+
+public static class ResourceGraphMapper
+{
+    private static readonly string s_databaseIcon = GetIconPathData(new Icons.Filled.Size24.Database());
+    private static readonly string s_containerIcon = GetIconPathData(new Icons.Filled.Size24.Box());
+    private static readonly string s_executableIcon = GetIconPathData(new Icons.Filled.Size24.SettingsCogMultiple());
+    private static readonly string s_projectIcon = GetIconPathData(new Icons.Filled.Size24.CodeCircle());
+
+    public static ResourceDto MapResource(ResourceViewModel r, IDictionary<string, ResourceViewModel> resourcesByName, IStringLocalizer<Columns> columnsLoc)
+    {
+        var resolvedNames = new List<string>();
+
+        foreach (var resourceRelationships in r.Relationships.GroupBy(r => r.ResourceName, StringComparers.ResourceName))
+        {
+            var matches = resourcesByName.Values
+                .Where(r => string.Equals(r.DisplayName, resourceRelationships.Key, StringComparisons.ResourceName))
+                .Where(r => r.KnownState != KnownResourceState.Hidden)
+                .ToList();
+
+            foreach (var match in matches)
+            {
+                resolvedNames.Add(match.Name);
+            }
+        }
+
+        var endpoint = ResourceEndpointHelpers.GetEndpoints(r, includeInternalUrls: false).FirstOrDefault();
+        var resolvedEndpointText = ResolvedEndpointText(endpoint);
+        var resourceName = ResourceViewModel.GetResourceName(r, resourcesByName);
+        var color = ColorGenerator.Instance.GetColorHexByKey(resourceName);
+
+        var icon = r.ResourceType switch
```

Comment:
We're going to want to allow the resource to specify this.

Reply:
@glennc is exploring resource-provided icons